game-core/src/main/java/games/strategy/util/ExitStatus.java

```diff
@@ -0,0 +1,27 @@
+package games.strategy.util;
+
+/**
+ * A process exit status.
+ */
+public enum ExitStatus {
+  /** The process exited successfully (0). */
+  SUCCESS(0),
+
+  /** The process exited due to a failure (1). */
+  FAILURE(1);
+
+  private final int status;
+
+  private ExitStatus(final int status) {
```

Comment:
nit: static analysis will nit that the `private` modifier is unnecessary on enum constructors, it's implicitly private. My 2 cents, `@AllArgsConstructor` saves a bit of syntax, the assert is not really necessary given we can see/control all values and it's pretty trivially true. I think the non-negative enforcement/commentary is overkill and perhaps has too much historical context compared to what a new reader of the code would see and note.

Reply:
> the `private` modifier is unnecessary on enum constructors

Roger that.

> the assert is not really necessary given we can see/control all values and it's pretty trivially true

Normally, I would agree, but it apparently wasn't that clear to whoever used exit status `-1` all over the place. :smile:  I was simply guarding against someone, a year from now, adding a new enum element with a negative status.  But, I guess the "pattern" of having a positive value for `FAILURE` might be enough to convey that requirement in the future.

<hr>

#3695